server/src/main/java/io/crate/protocols/postgres/types/BitType.java

```diff
@@ -93,12 +93,27 @@ public BitString readBinaryValue(ByteBuf buffer, int valueLength) {
     @Override
     byte[] encodeAsUTF8Text(BitString value) {
         assert length >= 0 : "BitType length must be set";
-        return value.asBitString().getBytes(StandardCharsets.UTF_8);
+        // See `varbit_out` in src/backend/utils/adt/varbit.c of PostgreSQL
+        return value.asRawBitString().getBytes(StandardCharsets.UTF_8);
     }
 
     @Override
     BitString decodeUTF8Text(byte[] bytes) {
         String text = new String(bytes, StandardCharsets.UTF_8);
-        return BitString.ofBitString(text);
+        // See `varbit_in` in src/backend/utils/adt/varbit.c of PostgreSQL
+        // PostgreSQL also supports hex notation, that's currently unsupported.
+
+        // bit-or 32 normalizes ascii-upper-case:
+        // "Bar".charAt(0) | 32 == 'b'
+        // "bar".charAt(0) | 32 == 'b'
+        if (!text.isEmpty() && (text.charAt(0) | 32) == 'b') {
+            return BitString.ofBitString(text);
```

Comment:
Why do we ignore `length` in this case, and we now use `text.lenght()` which contains also the `B'` prefix.

Reply:
Mostly because that's how the constructors currently work. 

I'm not sure if B prefixed strings are allowed to have a length that doesn't match the type's length

Reply:
I didn't look any deeper, I'm just worried that we return a BitString with a length bigger than the type by `2`.